controllers/hazelcast/wanreplication_controller.go

```
@@ -165,6 +176,56 @@ func (r *WanReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	return updateWanStatus(ctx, r.Client, wan, wanSuccessStatus())
 }
 
+func (r *WanReplicationReconciler) checkConnectivity(ctx context.Context, req ctrl.Request, wan *hazelcastv1alpha1.WanReplication, logger logr.Logger) error {
+	for _, rr := range wan.Spec.Resources {
+		hzResourceName := rr.Name
+		if rr.Kind == hazelcastv1alpha1.ResourceKindMap {
+			m := &hazelcastv1alpha1.Map{}
+			nn := types.NamespacedName{
+				Namespace: req.NamespacedName.Namespace,
+				Name:      rr.Name,
+			}
+			if err := r.Get(ctx, nn, m); err != nil {
+				if kerrors.IsNotFound(err) {
+					logger.V(util.DebugLevel).Info("Could not find Map")
+				}
+				return err
+			}
+			hzResourceName = m.Spec.HazelcastResourceName
+		}
+
+		statusService, ok := r.statusServiceRegistry.Get(types.NamespacedName{
+			Namespace: req.Namespace,
+			Name:      hzResourceName,
+		})
+		if !ok {
+			logger.Error(errors.New("get Hazelcast Status Service failed"), "")
```

Should we return the error here instead of logging?

---

Done.